testkit/src/main/java/akka/kafka/testkit/internal/KafkaContainer.java

```diff
@@ -0,0 +1,179 @@
+    // internal
+    // listener when KAFKA_INTER_BROKER_LISTENER_NAME is set, otherwise Kafka will try to use the
+    // advertised listener
+    withEnv("KAFKA_LISTENERS", "PLAINTEXT://0.0.0.0:" + exposedPort + ",BROKER://0.0.0.0:9092");
+    withEnv("KAFKA_LISTENER_SECURITY_PROTOCOL_MAP", "BROKER:PLAINTEXT,PLAINTEXT:PLAINTEXT");
+    withEnv("KAFKA_INTER_BROKER_LISTENER_NAME", "BROKER");
+
+    withEnv("KAFKA_BROKER_ID", brokerId);
+    withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", internalTopicRf + "");
+    withEnv("KAFKA_OFFSETS_TOPIC_NUM_PARTITIONS", internalTopicRf + "");
+    withEnv("KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR", internalTopicRf + "");
+    withEnv("KAFKA_TRANSACTION_STATE_LOG_MIN_ISR", internalTopicRf + "");
+    withEnv("KAFKA_LOG_FLUSH_INTERVAL_MESSAGES", Long.MAX_VALUE + "");
+    withEnv("KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS", "0");
+  }
+
+  public KafkaContainer withEmbeddedZookeeper() {
+    externalZookeeperConnect = null;
+    return self();
+  }
+
+  public KafkaContainer withExternalZookeeper(String connectString) {
+    externalZookeeperConnect = connectString;
+    return self();
+  }
+
+  public String getBootstrapServers() {
+    if (port == PORT_NOT_ASSIGNED) {
+      throw new IllegalStateException("You should start Kafka container first");
+    }
+    return String.format("PLAINTEXT://%s:%s", getContainerIpAddress(), port);
+  }
+
+  @Override
+  protected void doStart() {
+    withCommand("sleep infinity");
+
+    if (externalZookeeperConnect == null) {
+      addExposedPort(ZOOKEEPER_PORT);
+    }
+
+    super.doStart();
+  }
+
+  @Override
+  protected void containerIsStarting(InspectContainerResponse containerInfo) {
+    super.containerIsStarting(containerInfo);
+
+    port = getMappedPort(this.exposedPort);
+
+    final String zookeeperConnect;
+    if (externalZookeeperConnect != null) {
+      zookeeperConnect = externalZookeeperConnect;
+    } else {
+      zookeeperConnect = startZookeeper();
+    }
+
+    String internalIp = containerInfo.getNetworkSettings().getIpAddress();
+
+    ExecCreateCmdResponse execCreateCmdResponse =
+        dockerClient
+            .execCreateCmd(getContainerId())
+            .withCmd(
+                "sh",
+                "-c",
+                ""
+                    + "export KAFKA_ZOOKEEPER_CONNECT="
+                    + zookeeperConnect
+                    + "\n"
+                    + "export KAFKA_ADVERTISED_LISTENERS="
+                    + getBootstrapServers()
+                    + ","
+                    + String.format("BROKER://%s:9092", internalIp)
+                    + "\n"
+                    + "/etc/confluent/docker/run")
+            .exec();
+
+    try {
+      dockerClient
+          .execStartCmd(execCreateCmdResponse.getId())
+          .exec(new ExecStartResultCallback())
+          .awaitStarted(10, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

Comment:
@ennru This was my crappy workaround for the Lombok sneakythrows usage in the upstream PR.  sneakythrows allows the user to not define checked exceptions (kind of like how Scala methods work), but if you don't have Lombok available at compile time with your build tool then the sneakythrows attribute can't be used, which means the exception must be handled, which I can't add as a checked exception in this case or I wouldn't be able to override this method `containerIsStarting`.  Does that make sense?

Reply:
https://github.com/seglo/testcontainers-java/blob/seglo/multi-broker-kafka/modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainer.java#L95..L120

Reply:
Discussed offline.  Will rethrow `RuntimeException` with inner throwable.